### DISCUSSION TITLE
[New Feature] Temporary Mind Control Feature

### DISCUSSION THREAD

**Discussion by deathreaperz** [Phobos-developers/Phobos#888] [Ideas and feedback]
I just thought of this, so...

```
[A Warhead]
TemporaryMindControl= (bool)
TemporaryMindControl.Duration= (int)
TemporaryMindControl.Animation= (AnimList, such as MINDANIM)
TemporaryMindControl.TimerImage= (AnimList, such as IVANBOMB, can use same logic as Ivan Bomb)
TemporaryMindControl.TimerFlickerRate= (int, can use same logic as Ivan Bomb)
TemporaryMindControl.KillAfterDuration= (bool)
```

AFAIK, Warheads from SuperWeapon is neutral-owned, so...
```
[A SuperWeapon]
Type=TemporaryMindControl
TemporaryMindControl.Duration= (int)
TemporaryMindControl.Animation= (AnimList, such as MINDANIM)
TemporaryMindControl.TimerImage= (AnimList, such as IVANBOMB, can use same logic as Ivan Bomb)
TemporaryMindControl.TimerFlickerRate= (int, can use same logic as Ivan Bomb)
TemporaryMindControl.KillAfterDuration= (bool)
SW.AITargeting=PsychicDominator
_Support other SW Settings as well..._
```

**Comment by Otamaa** [comment]
It is possible and the game wil count the time per frame , check code below 
i dont want to mess up with git since i got limited internet atm :  

Warhead Code at BulletDetonate : 

if (pWHExt->MCTimer > 0) { //MCTImer is int
	if (pWH->MindControl && !pWP->InfiniteMindControl) {
		if (auto pTarget = abstract_cast<TechnoClass*>(pThis->Target)) {
					auto pExt = TechnoExt::ExtMap.Find(pTarget);
                                        //Or just use . CaptureTime.Start(pWHExt->MCTimer);
					pExt->CaptureTime.TimeLeft = pWHExt->MCTimer;
					pExt->CaptureTime.StartTime = Unsorted::CurrentFrame;
				}
			}
		}
TechnoClass Ext add : 
TimerStruct CaptureTime

Hook , TechnoClass Update :
//code here is copy paste from mind control range 

	if (auto Capturer = pThis->MindControlledBy) {
		auto pExt = TechnoExt::ExtMap.Find(pThis);
		if (pExt->CaptureTime.Completed()){
			Capturer->CaptureManager->FreeUnit(pThis);
			pExt->CaptureTime.Stop();
		if (!pThis->IsHumanControlled) {
				pThis->QueueMission(Mission::Hunt, 0);
			};
		}
	}
	
For animation and stuffs , can be added later on , i suppose 
this code  just proof of concept that the thing is possible to do

**Comment by secsome** [comment]
@Otamaa now you can edit it easier since the **Improved MindControl** has been introduced
